ui/src/main/webapp/src/app/configuration/configuration.component.ts

```diff
@@ -249,4 +254,49 @@ export class ConfigurationComponent implements OnInit, AfterViewInit {
     isFilterActive() {
         return this.filter.selectedFilters.length > 0;
     }
+
+    clickHeader(event:Event, provider:RuleProviderEntity) {
+        if(!$(event.target).is("button, a, input, .fa-ellipsis-v")){
+            $(this._element.nativeElement).find("#span-" + provider.id).toggleClass("fa-angle-down")
+                .end().parent().toggleClass("list-view-pf-expand-active")
+                .find("#container-" + provider.id).toggleClass("hidden").end().parent()
+                .find("#group-item-" + provider.id).toggleClass("selectedRuleHeader");
+            prettyPrint();
+        }
+    }
+
+    scrollToRule(id:number) {
+        this.scrollToElement(this._element.nativeElement.querySelector(`h4[id="${id}"]`));
+    }
+
+    scrollToRuleSetHeader(id:number) {
+        this.scrollToElement(this._element.nativeElement.querySelector(`div[id="group-item-${id}"]`));
+    }
+
+    private scrollToElement(element:Element) {
+        if (element) {
+            /*
+             * For reference on how the offset is computed:
+             * https://developer.mozilla.org/en/docs/Web/API/Element/getBoundingClientRect
+             *
+             * 60 is the height in px of the top nav bar "header-logo-wrapper"
+             * */
+            let offset = element.getBoundingClientRect().top + window.scrollY - 60;
+            window.scrollTo(0, offset);
+        }
+    }
+
+    getSpanAngleStyle(ruleProvider: RuleProviderEntity) {
+        let margin = 0;
+        if (ruleProvider.sources.length > 0) {
+            if (ruleProvider.targets.length > 0) margin = 14;
+            else margin = 7;
+        } else if (ruleProvider.targets.length > 0) margin = 7;
+        return this._sanitizer.bypassSecurityTrustStyle("margin-top: " + margin + "px;");
```

Comment:
I think yes because if i just use:

`return "margin-top: " + margin + "px;";`

the style is not applied and in the dev tools console i see a lot of messages like:
`WARNING: sanitizing unsafe style value margin-top: 14px;`